README.md

```diff
@@ -75,6 +75,30 @@ This function can also be used in hiera files, for example to set class paramete
 some_class::password: "%{lookup('important-secret')}"
 ```
 
+You can use a fact to specify different vaults for different groups of nodes, for example:
+
+```yaml
+- name: 'Azure Key Vault Secrets per environment'
+    lookup_key: azure_key_vault::lookup
+    options:
+      vault_name: "%{facts.environment}-vault"
```

Comment:
Lets use trusted.extensions.pp_environment here since using environment is a bad practice since puppet environments don't always correlate to an organizations environment.

Reference:
https://github.com/puppetlabs/best-practices/blob/master/use-of-environment-in-hiera-hierarchy.md